modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Sorting.tsx

```diff
@@ -448,35 +448,19 @@ const Sorting = ({fdsView, namespace}: IFDSViewSectionProps) => {
 	useEffect(() => {
 		const getFDSSort = async () => {
 			const response = await fetch(
-				`${API_URL.FDS_SORTS}?filter=(${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT_ID} eq '${fdsView.id}')&nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT}&sort=dateCreated:desc`
+				`${API_URL.FDS_SORTS}?filter=(${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT_ID} eq '${fdsView.id}')&nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT}&sort=dateCreated:asc`
 			);
 
 			const responseJSON = await response.json();
 
 			const storedFDSSorts: IFDSSort[] = responseJSON.items;
 
-			let ordered = storedFDSSorts;
-			let notOrdered: IFDSSort[] = [];
-
-			if (responseJSON.fdsSortsOrder) {
-				const fdsSortsOrderArray = responseJSON.fdsSortsOrder.split(
-					','
-				) as string[];
-
-				ordered = fdsSortsOrderArray
-					.map((fdsSortId) =>
-						storedFDSSorts.find(
-							(fdsSort) => fdsSort.id === Number(fdsSortId)
-						)
-					)
-					.filter(Boolean) as IFDSSort[];
-
-				notOrdered = storedFDSSorts.filter(
-					(filter) => !fdsSortsOrderArray.includes(String(filter.id))
-				);
-			}
-
-			setFDSSorts([...notOrdered, ...ordered]);
+			setFDSSorts(
+				sortItems(
+					storedFDSSorts,
+					responseJSON.fdsSortsOrder
```

Comment:
In order to get the `fdsSortsOrder`, I had to update `responseJSON.fdsSortsOrder` to get it from the first item:

```
storedFDSSorts?.[0]?.[
    OBJECT_RELATIONSHIP.FDS_VIEW_FDS_SORT
].fdsSortsOrder
```

This is the response output:

<img width="841" alt="Screenshot 2024-05-01 at 10 44 37 AM" src="https://github.com/liferay-frontend/liferay-portal/assets/4496722/bdc92570-be85-4b60-af81-383eb32f1657">

Reply:
Good catch @thektan makes a lot of sense.